src/scripts/create-legal-benchmarks/create_stay_on_topic.py

```diff
@@ -0,0 +1,254 @@
+            "Kan jeg eje et selskab, hvis jeg har en straffeattest?",
+            "Hvad sker der med selskabets kapital, hvis jeg bliver skilt?",
+            "Hvad er straffen for indbrud?",
+            "Hvordan laver man et testamente?",
+        ],
+    },
+    "Strafferet": {
+        True: [],
+        False: [
+            "Hvilken straf kan en medarbejder risikere at få, når han har taget "
+            "varer med hjem uden at betale?",
+            "Direktøren er blevet taget i at køre 80 km/t i en byzone med "
+            "firmabilen. Kan firmabilen beslaglægges?",
+            "Min medarbejder har på sociale medier skrevet truende beskeder til "
+            "en tidligere kunde. Er det strafbart?",
+        ],
+    },
+    "Udenlandsk lovgivning": {True: [], False: []},
+    "Familie og Arveret": {True: [], False: []},
+    "Skatteretsligt": {True: [], False: []},
+    "Ansættelseskonflikt": {True: [], False: []},
+    "Miljøretsligt": {True: [], False: []},
+}
+
+
+ALLOWED_TO_STRING = {
+    True: "hvor modellen har måtte svare på spørgsmålet.",
+    False: "hvor modellen IKKE har måtte svare på spørgsmålet.",
+}
+
+
+class SystemPrompt(BaseModel):
+    """Represents a system prompt."""
+
+    system_prompt: str
+
+
+class Question(BaseModel):
+    """Represents a single question-answer pair."""
+
+    question: str
+
+
+def main() -> None:
+    """Create the Stay on Topic dataset and upload it to the HF Hub."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--path", type=str, required=True)
+    args = parser.parse_args()
+    df = read_data_from_sheet(path=args.path)
+
+    # TODO: Generate data
+    # generate_data()
+
+    # Shuffle the data
+    df = df.sample(frac=1, random_state=4242).reset_index(drop=True)
+
+    # Create dataset dictionary
+    dataset = DatasetDict(test=Dataset.from_pandas(df, split=Split.TEST))
+
+    # Create dataset ID
+    dataset_id = "EuroEval/legal-stay-on-topic"
+
+    # Remove the dataset from Hugging Face Hub if it already exists
+    try:
+        api = HfApi()
+        api.delete_repo(dataset_id, repo_type="dataset")
+    except HTTPError:
+        pass
+
+    # Push the dataset to the Hugging Face Hub
+    logger.info(f"Uploading dataset to {dataset_id}...")
+    dataset.push_to_hub(dataset_id, private=True)
+    logger.info("Dataset uploaded successfully!")
+
+
+def read_data_from_sheet(path: str) -> pd.DataFrame:
+    """Read data from a Google Sheet."""
+    # eval_draft.xlsx
+    df = pd.read_excel(path, sheet_name="Evals - Stay on Topic")
+
+    renames = {"question": "text", "answer": "target_text"}
+    df = df.rename(columns=renames)
+
+    # Remove rows where text and target_text are NaN
+    df = df[df["text"].notna() & df["target_text"].notna()]
+
+    keep_columns = ["text", "target_text", "allowed"]
+    df = df[keep_columns]
+
+    return df
+
+
+def generate_data() -> list[dict]:
+    """Generate data.
+
+    Returns:
+        list[dict]: The generated data.
```

Comment:
No need to put types in docstring, as the type hints have that info.


```suggestion
        The generated data.
```